File: LL/src/Core/Application/UseCases/CharacterActions/Dtos/CombatDtos/CombatResultDto.cs
using Application.Common.Mappings;
using AutoMapper;
using Domain.Models.Combat;
using Domain.Models.Inventories;

namespace Application.UseCases.CharacterActions.Dtos.CombatDtos;
public class CombatResultDto : IMapFrom<CombatResult>
{
    public List<CombatEntityDto> PlayerTeam { get; set; } = [];
    public List<CombatEntityDto> EnemyTeam { get; set; } = [];
    public List<CombatEvent> EventLog { get; set; } = [];
    public BattleOutcome Outcome { get; set; }
    public List<InventoryItem> Loot { get; set; } = [];
    public int ExperienceGained { get; set; }
    public DateTimeOffset StartedAt { get; set; }
    public int Duration { get; set; }

    public void Mapping(Profile profile)
    {
        profile.CreateMap<CombatResult, CombatResultDto>();
    }
}

File: LL/src/Presentation/dashboard/src/app/shared/models/Dtos/combatResultDto.ts
import { CombatEvent } from './combatEventDto';

export interface CombatResultDto {
  playerTeam: SimpleCombatEntityDto[];
  enemyTeam: SimpleCombatEntityDto[];
  duration: number;
  eventLog: CombatEvent[];
  startedAt: Date;
  outcome: BattleOutcome;
  experienceGained: number;
}

export interface SimpleCombatEntityDto {
  name: string;
  id: string;
  imagePath: string;
  health: number;
  maxHealth: number;
  barrier: number;
}

export enum BattleOutcome {
  Victory = 'Victory',
  Defeat = 'Defeat',
  Draw = 'Draw',
}
